parsers/mysql/mysql.go

```diff
@@ -260,20 +267,23 @@ func (p *Parser) ProcessLines(lines <-chan string, send chan<- event.Event) {
 	go p.handleEvents(rawEvents, send)
 
 	// flag to indicate when we've got a complete event to send
-	var sendEvent bool
+	var foundStatement bool
 	groupedLines := make([]string, 0, 5)
 	for line := range lines {
-		if strings.HasPrefix(line, "# ") && len(groupedLines) > 0 {
+		if foundStatement && len(groupedLines) > 0 && strings.HasPrefix(line, "# ") {
```

Comment:
can it ever be the case that `foundStatement == true` but `len(groupedLines) <= 0`?  It doesn't look like it.